components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -618,6 +542,16 @@ void AdsServiceImpl::Shutdown() {
   bat_ads_.reset();
   bat_ads_client_binding_.Close();
 
+  for (auto it = notifications_.cbegin(); it != notifications_.cend(); ++it) {
```

Comment:
we actually don't want to do this and it's probably why you're not seeing any cached notifications